__test__/api/replies/getSingleReplyVotes.test.js

```diff
@@ -0,0 +1,153 @@
+const supertest = require("supertest");
+const request = supertest(app);
+
+let comment, totalVotes;
+let reply1, reply2;
+let savedReply1, savedReply2;
+
+describe("GET /comments/:commentId/replies/:replyId", () => {
+  beforeEach(async () => {
+    // Mock a comment document
+    const mockedCommentDoc = new CommentModel({
+      content: "A comment from user 1",
+      ownerId: "user1@email.com",
+      origin: "123123",
+      refId: 2,
+      applicationId: global.application._id,
+    });
+
+    // Mock a reply document.
+    const mockedReply1Doc = new ReplyModel({
+      content: "A reply from user 2",
+      ownerId: "user2@email.com",
+      commentId: mockedCommentDoc.id,
+      upVotes: [],
+      downVotes: [],
+    });
+
+    // Mock a reply document.
+    const mockedReply2Doc = new ReplyModel({
+      content: "A reply from user 3",
+      ownerId: "user3@email.com",
+      commentId: mockedCommentDoc.id,
+      flags: [mockedCommentDoc.ownerId],
+      upVotes: [],
+      downVotes: [mockedCommentDoc.ownerId],
+    });
+
+    // Save mocked comment document to the database and cache.
+    comment = await mockedCommentDoc.save();
+
+    //Mock voting
+    mockedReply1Doc.upVotes.push("newvoter@gmail.com");
+    mockedReply1Doc.downVotes.push("anothervoter@gmail.com");
+    totalVotes = mockedReply1Doc.upVotes.concat(mockedReply1Doc.downVotes);
+
+    mockedReply2Doc.upVotes.push("chivoter@gmail.com");
+    mockedReply2Doc.downVotes.push("avoter@gmail.com");
+
+    // Save mocked replies document to the database.
+    savedReply1 = await mockedReply1Doc.save();
+    savedReply2 = await mockedReply2Doc.save();
+
+    // Add replies to the mocked comment.
+    await CommentModel.findByIdAndUpdate(comment.id, {
+      $push: {
+        replies: { $each: [savedReply1.id, savedReply2.id] },
+      },
+    });
+
+    // Cache response objects
+    reply1 = replyHandler(savedReply1);
+    reply2 = replyHandler(savedReply2);
+  });
+
+  afterEach(async () => {
+    // Delete mocks from the database.
+    await CommentModel.findByIdAndDelete(comment.id);
+    await ReplyModel.findByIdAndDelete(reply1.replyId);
+    await ReplyModel.findByIdAndDelete(reply2.replyId);
+
+    // Delete cache.
+    comment = null;
+    reply1 = null;
+    reply2 = null;
+  });
+
+  it("should get a single reply and all its votes", async () => {
+    const url = `/v1/comments/${comment.id}/replies/${savedReply1.id}`;
+    const bearerToken = `bearer ${global.appToken}`;
+
+    const res = await request.get(url).set("Authorization", bearerToken);
+
+    expect(res.status).toEqual(200);
+    expect(res.body.status).toEqual("success");
+    expect(res.body.data).toMatchObject(reply1);
+    expect(res.body.data.numOfUpVotes[0]).toEqual(reply1.numOfUpVotes[0]);
+    expect(res.body.data.numOfDownVotes[0]).toEqual(reply1.numOfDownVotes[0]);
+    expect(res.body.data.numOfUpVotes + res.body.data.numOfDownVotes).toEqual(
+      totalVotes.length
+    );
```

Comment:
This is not what the endpoint returns and it should not be what we expect. See the swagger docs for more information on this.